ndscan/fitting/fitting.py

```diff
@@ -171,6 +171,31 @@ def residuals(self):
         """Returns the fit residuals."""
         return self._y - self._func_free(self._x, self._p)
 
+    @property
+    def chi2(self) -> Tuple[float, float]:
+        """ Chi-Squared test.
+
+        Assumes the values of `y` are independent and normally distributed.
+        Returns the tuple (Chi-Squared, p-value)
+
+        The the p-value is the probability that the observed deviations from the fit
+        could be observed by chance assuming the data is normally distributed (values
+        close to unity indicate high-quality fits).
+        """
+        if self._y_err is None:
+            raise ValueError("Cannot calculate chi square without knowing y_err")
+
+        n = len(self._x) - len(self._free_params)
+        if n < 1:
+            raise ValueError("Cannot calculate chi squared with "
+                             f"{len(self._free_params)} fit parameters and only "
+                             f"{len(self._x)} data points.")
+
+        y_fit = self.evaluate(self._x)[1]
+        chi_2 = np.sum(np.power((self._y - y_fit) / self._y_err, 2))
+        p = stats.chi2.sf(chi_2, n)
```

Comment:
Todo: check the standard definition of `p` (is it cdf or sf? i.e. if all points lie on the curve is p 1 or 0).